swift_browser_ui_frontend/src/common/store.js

```diff
@@ -145,78 +142,90 @@ const store = new Vuex.Store({
     },
   },
   actions: {
-    updateContainers: async function ({ commit, dispatch }, signal) {
+    updateContainers: async function (
+      { commit, dispatch, state },
+      {route, signal},
+    ) {
       commit("loading", true);
-      let containers = [];
-      await getBuckets().then((ret) => {
-        if (ret.status != 200) {
+      let project = route.params.project;
+      commit("updateContainers", []);
+      let containers;
+      let marker = "";
+      do {
+        containers = [];
+        containers = await getBuckets(
+          project,
+          marker,
+        ).catch(() => {
           commit("loading", false);
-        }
-        containers = ret;
-        commit("updateContainers", ret);
-        commit("loading", false);
-      }).catch(() => {
+        });
         commit("loading", false);
+        if (containers.length > 0) {
+          commit("updateContainers", state.containerCache.concat(containers));
+          marker = containers.pop().name;
+        }
+      } while (containers.length > 0);
+      containers = state.containerCache;
+      dispatch("updateContainerTags", {
+        project,
+        containers,
+        signal,
       });
-      dispatch("updateContainerTags", {containers, signal});
       return containers;
     },
-    updateContainerTags: function ({ commit }, {containers, signal}) {
-      containers.map(async container => {
-        const tags = await getTagsForContainer(container.name, signal);
-        commit(
-          "updateContainerTags", 
-          {containerName: container.name, tags},
-        );
-      });
+    updateContainerTags: function (
+      { commit },
+      {project, containers, signal},
+    ) {
+      if (containers.length < 10000) {
```

Comment:
Is there any case of people with that many containers? We would need many other optimizations in that case.

Reply:
As far as I know it's a use case we should and need to support. Not unfeasible to expect >10k items in a container. But for now I think not being able to provide all the nice things we have added on top of the Openstack APIs when people reach the limits of the storage backend is a reasonable compromise.